### DISCUSSION TITLE
facetedDistribution to returns topHits

### DISCUSSION THREAD

**Discussion by georgesve** [meilisearch/product#700] [Feedback & Feature Proposal]
**What**
I would like the factedDistribution to not only return the count of matching documents but also a limited selection of matching documents. 

**Why**
This would be particularly useful in a catalogue to display the first matching documents of each facets (ex: display first 5 products of each categories or the first 5 products of each brands)

**Current solution**
Use multi_search and filters, but it's not great as we need to grab all the facets before executing the multi_search.

**Expected results**
What would be great is to have something like this
`
"facetDistribution": {
      "categories": {
        "toys": {
          "count": 5,
          "topHits": [
            { "id": 1, "name": "Toy A", "category": "toys", "price": "$10" },
            { "id": 2, "name": "Toy B", "category": "toys", "price": "$12" }
          ]
        },
        "books: {
          "count": 1, 
          "topHits": [
            { "id": 1, "name": "Book A", "category": "books", "price": "$10" }
          ]
        }
      }
    }
   `
   
<img width="702" alt="Capture d’écran 2023-10-12 à 11 10 40" src="https://github.com/meilisearch/product/assets/8372854/da863a82-b09a-4bf5-a471-6f715d0bddce">

**Comment by macraig** [comment]
Hi @georgesve, thank you for submitting your feedback and describing your use case with the workaround included :)  I'll keep this discussion open for other users to upvote or add their own use cases to it.
